### PR TITLE
Update docs and pull request template

### PR DESCRIPTION
**Pull request by wlach**
Notably, no need to ask people to run the integration tests
manually since we now do this automatically. The instructions
were out of date and hard to follow so I just removed them, if
people really care to do this themselves they can always look at the
circle configuration.

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant
- [ ] Scan the PR and verify that no changes (particularly to `.circleci/config.yml`) will cause environment variables (particularly credentials) to be exposed in test logs
- [ ] Trigger the `integration` CI test by pushing this revision [as discussed in the README](https://github.com/mozilla-services/mozilla-pipeline-schemas#packaging-and-integration-tests-optional) and review the report posted in the comments.

For glean changes:
- [ ] Update `include/glean/CHANGELOG.md`

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 Checklist for reviewer:
 
-- [ ] Commits should reference a bug or github issue, if relevant
+- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
 - [ ] Scan the PR and verify that no changes (particularly to `.circleci/config.yml`) will cause environment variables (particularly credentials) to be exposed in test logs
-- [ ] Trigger the `integration` CI test by pushing this revision [as discussed in the README](https://github.com/mozilla-services/mozilla-pipeline-schemas#packaging-and-integration-tests-optional) and review the report posted in the comments.
+- [ ] Watch the results of the `integration` CI test and make sure the error rates do not increase without justification.
```
Comment (acmiyaguchi) [inline .github/pull_request_template.md]: The integration tests may still need to be triggered manually if the PR comes from a fork of the repository.

**Comment by jklukas** [comment]
Same comments as @acmiyaguchi. The reviewer still has to intervene to kick off integration tests on forks.

File: .github/pull_request_template.md (hunk 1 above)
Comment (wlach) [reply .github/pull_request_template.md]: I'll revert this change and make a note about the fork part.
